src/test/scala/io/iohk/ethereum/blockchain/sync/SyncControllerSpec.scala

```diff
@@ -511,6 +511,78 @@ class SyncControllerSpec extends AnyFlatSpec with Matchers with BeforeAndAfter w
         }
       }
     }
+    // scalastyle:off method.length
+    case class SyncStateAutoPilot(
+        handshakedPeers: HandshakedPeers,
+        pivotHeader: BlockHeader,
+        blockchainData: BlockchainData,
+        failedReceiptsTries: Int,
+        failedBodiesTries: Int,
+        onlyPivot: Boolean,
+        failedNodeRequest: Boolean
+    ) extends AutoPilot {
+      override def run(sender: ActorRef, msg: Any): AutoPilot = {
+        msg match {
+          case EtcPeerManagerActor.GetHandshakedPeers =>
+            sender ! handshakedPeers
+            this
+
+          case SendMessage(msg: GetBlockHeadersEnc, peer) =>
+            val underlyingMessage = msg.underlyingMsg
+            if (underlyingMessage.maxHeaders == 1) {
+              // pivot block
+              sender ! MessageFromPeer(BlockHeaders(Seq(pivotHeader)), peer)
+              this
```

Comment:
Minor: You don't mutate state, so you could move `this` after the if statement. It will simplify a logic a little bit